#! /bin/sh -e

# DP: vboxd fix for little-endian machines.

dir=
if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
    dir="$3/"
elif [ $# -ne 1 ]; then
    echo >&2 "usage: `basename $0`: -patch|-unpatch [-d <srcdir>]"
    exit 1
fi
case "$1" in
    -patch)
        patch $pdir -f --no-backup-if-mismatch -p0 < $0
        ;;
    -unpatch)
        patch $pdir -f --no-backup-if-mismatch -R -p0 < $0
        ;;
    *)
	echo >&2 "usage: `basename $0`: -patch|-unpatch [-d <srcdir>]"
        exit 1
esac
exit 0

--- vbox/src/vboxd.c~	2005-03-28 19:50:15.770709237 +0200
+++ vbox/src/vboxd.c	2005-03-28 19:58:56.780069971 +0200
@@ -387,7 +387,7 @@
 	fd_set         rmask;
 	char          *stop;
 	int            p;
-	int            c;
+	char           c;
 	int            rc;
 
 	*line = '\0';
